hyperspy/_components/exponential.py

```diff
@@ -149,10 +146,13 @@ def estimate_parameters(self, signal, x1, x2, only_current=False):
                                 'a zero or negative value).')
                 return False
 
-            if is_binned(signal) is True:
+            if is_binned(signal):
             # in v2 replace by
             #if axis.is_binned:
-                A /= axis.scale
+                if axis.is_uniform:
+                    A /= axis.scale
+                else:
+                    A /= np.gradient(axis.axis)[i_mid]
```

Comment:
Just wondering about the use of `i_mid` here. Is that essentially a "best guess" for the bin width? Have we considered other approaches here?

Reply:
Yes, that was my idea of a "best guess" in the parameter estimator as we have no idea how the axis evolves. For `Gaussian` and the like I took the `centre` position, but `Exponential` and `Polynomial` do not have that, so I resorted to this guess. If anyone has a better idea, I'm happy to take that up, but for the estimator it should not get too complicated either I would say. An alternative could be `np.mean(np.gradient(...))` instead.

Reply:
I think that's fine. But I suggest you add a few short comments explaining that this is a best guess, perhaps with your suggested alternative. I think you should do this (copy paste) on each component where you use such a guess. I'm just imagining someone trying to decipher this in the future where perhaps you are unavailable :)